### PR TITLE
KEYCLOAK-14536 Add upgrading guide pointing out UserModel changes

### PR DESCRIPTION
**Pull request by Martin-Idel-SI**
Add information about the changes within `UserModel` and describe who could be affected and how to mitigate.

Connects to https://github.com/keycloak/keycloak/pull/7080

### CODE REVIEW

**Review by mposolda — DISMISSED** [review]
@Martin-Idel-SI Thanks!

@andymunro Do you please have a chance to review this PR?

File: upgrading/topics/keycloak/changes.adoc [hunk 1]
```diff
@@ -12,6 +12,17 @@ This should not affect most of the deployments, but some can be affected under s
 tried to update a user with the admin REST API and the user had some incorrect attribute changes, the update was possible. With the
 current version, the update is not possible and you will be immediately informed about the reason.
 
+==== UserModel changes
+
+The fields `username`, `email`, `firstName` and `lastName` in the `UserModel` are migrated to custom attributes as a preparation for adding more sophisticated user profiles to {project_name} in an upcoming version.
+If a database contains users with custom attributes of that exact name, the custom attributes will need to be migrated before upgrading (this is not done automatically). Otherwise, they will not be read from the database anymore and possibly deleted.
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: Change " upgrading (this is not done automatically)"

to  "upgrading. This migration does not occur automatically."

File: upgrading/topics/keycloak/changes.adoc [hunk 2]
```diff
@@ -12,6 +12,17 @@ This should not affect most of the deployments, but some can be affected under s
...
+This implies, that the `username` can now also be accessed and set via `UserModel.getFirstAttribute(UserModel.USERNAME)` (and similar for the other fields).
```
Comment (andymunro) [inline upgrading/topics/keycloak/changes.adoc]: Change "This implies" to "This situation implies"
Change "(and similar for the other fields)." to "Similar implications exist for other fields."

**Review by andymunro — CHANGES_REQUESTED** [review]
Thanks, Marek.  A couple of minor changes.

**Comment by Martin-Idel-SI** [comment]
@andymunro Thanks, I always get carried away with using brackets when they are hardly ever appropriate... :-D

Updated with your suggestions.

**Review by mposolda — APPROVED** [review]
@andymunro @Martin-Idel-SI Thanks for the review and updates!

**Pull request merged**